crates/context_aware_config/src/api/dimension/handlers.rs

```diff
@@ -83,40 +105,87 @@ async fn create(
         last_modified_by: user.get_email(),
     };
 
-    let upsert = diesel::insert_into(dimensions)
-        .values(&new_dimension)
-        .on_conflict(dimensions::dimension)
-        .do_update()
-        .set(&new_dimension)
-        .get_result::<Dimension>(&mut conn);
-
-    match upsert {
-        Ok(upserted_dimension) => {
-            let is_mandatory = tenant_config
-                .mandatory_dimensions
-                .contains(&upserted_dimension.dimension);
-            Ok(HttpResponse::Created().json(DimensionWithMandatory::new(
-                upserted_dimension,
-                is_mandatory,
-            )))
-        }
-        Err(diesel::result::Error::DatabaseError(
-            diesel::result::DatabaseErrorKind::ForeignKeyViolation,
-            e,
-        )) => {
-            log::error!("{fun_name:?} function not found with error: {e:?}");
-            Err(bad_argument!(
-                "Function {} doesn't exists",
-                fun_name.unwrap_or(String::new())
-            ))
-        }
-        Err(e) => {
-            log::error!("Dimension upsert failed with error: {e}");
-            Err(unexpected_error!(
-                "Something went wrong, failed to create/update dimension"
-            ))
+    conn.transaction::<_, superposition::AppError, _>(|transaction_conn| {
+        match (prev_index, new_index.clone(), req_position_val.clone()) {
+            (Some(prev_val), Some(new_val), None) => {
+                println!("1");
+                if prev_val < new_val {
+                    new_dimension.position = new_val.clone() as i32 - 1;
```

Comment:
Why are we doing this check? @pratikmishra356 can we add some comments here and use better variable names to document whats happening?

Reply:
Sorry , missed this will have to write comments in code to explain this